keyboards/keaboard/rev1/rev1.h

```diff
@@ -0,0 +1,35 @@
+/*
```

Comment:
This file, rev1.c and keaboard.[ch] can all be deleted if you add matrix positions to the layout in info.json: https://docs.qmk.fm/#/reference_info_json?id=layout-format

Reply:
Well, I'm part way there. It compiles but I definitely got my matrix wrong lol

Reply:
I'm currently at a loss trying to understand how the new `info.json` matrix mapping works. The documentation is exceedingly light and I'm having a hard time finding any existing examples of what you're asking for.

I got my first row on the left half mapped correctly but I'm clearly missing something in my understanding as continuing what I thought was the pattern to the second row is instead continuing on across my keymap instead of dropping down to the second row.

Can you point me towards any other relevant docs or some keyboards configured with the new `info.json` matrix pattern? Ideally a split keyboard as I'm having trouble with the halves as well.

Reply:
I think I finally got it. Found the `keyboards/cantor` config which is similar to mine. While I still can't say I understand how the matrix works, I was able to mimic the Cantor config and stumble my way through to a matrix that works with my keymap. Pretty sure I even preserved the correct mappings to the `x`/`y` values. :relieved:

Reply:
The ordering isn't quite correct:
![image](https://user-images.githubusercontent.com/4781841/218385543-14f3149f-343d-4a30-8377-1aa9e4f63ee7.png)
You can test this by turning on fast input in the Configurator and then pressing Ctrl+Shift+I.

Reply:
As for how the matrix positions work, it's a little more difficult for the layout macro you made due to how you labelled the keys. Here's something which might make more sense, and is how the majority of layout macros still in the repo are:
```c
#define LAYOUT( \
    L00, L01, L02, L03, L04, L05,                 R05, R04, R03, R02, R01, R00, \
    L10, L11, L12, L13, L14, L15,                 R15, R14, R13, R12, R11, R10, \
    L20, L21, L22, L23, L24, L25, L26,       R26, R25, R24, R23, R22, R21, R20, \
    L30, L31, L32, L33, L34,      L36,       R36,      R34, R33, R32, R31, R30 \
) { \
    { L00, L01, L02, L03, L04, L05, XXX }, \
    { L10, L11, L12, L13, L14, L15, XXX }, \
    { L20, L21, L22, L23, L24, L25, L26 }, \
    { L30, L31, L32, L33, L34, XXX, k36 }, \
\
    { R00, R01, R02, R03, R04, R05, XXX }, \
    { R10, R11, R12, R13, R14, R15, XXX }, \
    { R20, R21, R22, R23, R24, R25, R26 }, \
    { R30, R31, R32, R33, R34, XXX, R36 } \
}
```

Reply:
Am I understanding correctly then that the `layout` array is analogous to the first half of your code snippet and the `matrix` property is defining a zero-indexed `[x,y]` location in the second half of the code snippet?

Reply:
> The ordering isn't quite correct: [image] You can test this by turning on fast input in the Configurator and then pressing Ctrl+Shift+I.

Oh, I see. Yep I "fixed" the layout previously by reordering lines so they mapped correctly when I should have instead only changed the matrix `y` configs.

I've got it correctly propagating with the configurator's fast input now and the mapping is confirmed correct with manual testing.